flow-server/src/main/java/com/vaadin/flow/component/UI.java

```
@@ -1195,6 +1195,28 @@ public boolean isNavigationSupported() {
         return true;
     }
 
+    /**
+     * Returns the currently active route aka navigation target shown in this
+     * UI.
+     * <p>
+     * Note, that certain UIs, like embedded apps, don't support routing and for
+     * those an empty optional is returned.
```

Javadoc needs to be updated, as now we throw an exception instead of return an Optional